metals/src/main/scala/scala/meta/internal/metals/FindTextInFiles.scala

```diff
@@ -0,0 +1,130 @@
+package scala.meta.internal.metals
+
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.nio.file.FileVisitResult
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+
+import scala.collection.mutable.ArrayBuffer
+import scala.util.control.NonFatal
+
+import scala.meta.internal.io.FileIO
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.lsp4j.Location
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+
+class FindTextInFiles(
+    buildTargets: BuildTargets,
+    workspace: () => AbsolutePath
+) {
+  def find(mask: String, content: String): List[Location] = {
+    if (mask == null || content == null) {
+      List.empty[Location]
+    } else {
+      val allLocations: ArrayBuffer[Location] = new ArrayBuffer[Location]()
+
+      buildTargets.allWorkspaceJars.foreach { classpathEntry =>
```

Comment:
I think we could cache text file paths when indexing, which would make it all much faster and wouldn't take up much more memory.

Reply:
So essentially an index of `jar path -> list of files inside a jar`? Would that not be huge (since there are lots of classfiles etc)?